packages/sudoku-game/src/lib/Game/Controller/components/Notes.svelte

```diff
@@ -1,64 +1,326 @@
+          if (row !== 0) {
+            newCell = { row: row - 1, column };
+          } else {
+            newCell = { row: 8, column };
+          }
+          break;
+        case 'right':
+          if (column !== dim.columns - 1) {
+            newCell = { row, column: column + 1 };
+          } else {
+            newCell = { row, column: 0 };
+          }
+          break;
+        case 'down':
+          if (row !== dim.rows - 1) {
+            newCell = { row: row + 1, column };
+          } else {
+            newCell = { row: 0, column };
+          }
+          break;
+        case 'left':
+          if (column !== 0) {
+            newCell = { row, column: column - 1 };
+          } else {
+            newCell = { row, column: 8 };
+          }
+          break;
+        default:
+          break;
+      }
+      if (newCell) {
+        k.preventDefault();
+        if (isCommandKey(k)) {
+          if ($selectedItemIndex > -1) {
+            addCellToSelectedNote(newCell);
+          } else {
+            selectedCells.add(newCell);
+          }
+        } else {
+          $selectedCells = [newCell];
+          $selectedItemIndex = -1;
+        }
+      }
+    }
+  };
+
+  onMount(() => {
+    $handleClickCell = customHandleClickCell;
+
+    $handleEnterCell = customHandleEnterCell;
+
+    $handleArrows = customHandleArrows;
+  });
 </script>
 
-<div class="w-full h-full flex flex-col justify-center items-center p-2">
-  <Input
-    bind:this={input}
-    asTextarea
-    bind:value
-    placeholder="note"
-    disabled={$selectedCells.length !== 1}
-    title={$selectedCells.length !== 1
-      ? 'Please select only one cell to make a note on'
-      : 'Write a note'}
-  />
-
-  <Button on:click={() => (value = '')} class="mt-2 w-full">Clear note</Button>
+<div class="grid grid-cols-2 w-full h-full p-2">
+  <div class="px-2 flex flex-col overflow-hidden justify-between">
+    <div
+      class="bg-gray-200 rounded-md shadow-inner flex flex-col items-center p-2 overflow-hidden h-full"
+    >
+      <div class="h-full overflow-y-auto w-full">
+        {#each $notes as note, index (index)}
+          <ControllerButton
+            isHighlighted={index === $selectedItemIndex}
+            onClick={() => {
+              selectedCells.set(note.positions);
+              $selectedItemIndex = index;
+            }}
+            onDelete={() => deleteNoteAtIndex(index)}
+            onHover={() => {
+              $highlightedCells = note.positions;
+              $highlightedItemIndex = index;
+            }}
+            onHoverOut={() => {
+              $highlightedCells = [];
+              $highlightedItemIndex = -1;
+            }}
+            onMoveUp={() => reorderNote(index, 'up')}
+            onMoveDown={() => reorderNote(index, 'down')}
+          >
+            <p
+              class={classNames(`text-${note.color != null ? note.color.toLowerCase() : 'black'}`)}
+            >
+              Note {index + 1}
```

Comment:
Perhaps display the content of the note here. i.e.
```suggestion
              Note {index + 1}: {note.details}
```

Reply:
I want to display the note in the button instead of an enumerated list, but just sticking the details in as text is not the solution, as it oveflows and stretches the button out. The commented out html below was my best attemnt at fixing that problem, but didn't work. The best solution would be one that displays the whole not, but wraps and scales it to fit in the buttons view box, like you do with the scaled svgs for the paths and cages in the editor control. Any suggestions on how to do that?

Reply:
What I did for paths and cages was to make a component for a single instance of a path. I then exported that from the sudoku-display package and imported it into the sudoku-game package where I could put it inside the <ScaledSvg /> component that takes care of the scaling and stuff